src/banman.h

```diff
@@ -37,6 +37,23 @@ class CSubNet;
 // dangerous, because it can cause a network split
 // between nodes running old code and nodes running
 // new code.
+//
+// With regards to impersonation attack, i.e a malicious
+// actor sending us directly invalid data while IP-spoofing
+// an address of honest reacheable peers, thus triggering an
+// unlawful ban of those:
+// * if attacker is a simple IP host, it won't work as our
+// default connections rely on TCP and being out-the-path
+// it will miss the SYN-ACK handshake reply.
+// * if attacker is an on-the-path infrastructure one (ISPs,
+// Tier2 or Tier1, ...), assuming IP packets forgeability
+// capabilities, a malicious packet can be injected, i.e
+// a mutated block received by node 1 from node 2 therefore
+// provoking ban of node 2 for 24h. Such threat model scenario
```

Comment:
Couldn't such attackers just block the connection ?

Reply:
Sure, an on-path attacker can drop the connection, but if honest peers have another connection path without him on it, that's still work. I would say banning is more harmful than a connection drop, and so more interesting ?